src/NServiceBus.Core/Causation/MessageCausation.cs

```diff
@@ -9,7 +9,23 @@ public MessageCausation()
 
         protected internal override void Setup(FeatureConfigurationContext context)
         {
-            context.Pipeline.Register("AttachCausationHeaders", new AttachCausationHeadersBehavior(), "Adds related to and conversation id headers to outgoing messages");
+            var newIdGenerator = GetIdGenerator(context);
+
+            context.Pipeline.Register("AttachCausationHeaders", new AttachCausationHeadersBehavior(newIdGenerator), "Adds related to and conversation id headers to outgoing messages");
+        }
+
+        static TryGetConversationIdDelegate GetIdGenerator(FeatureConfigurationContext context)
+        {
+            if (context.Settings.TryGet<TryGetConversationIdDelegate>(out var idGenerator))
+            {
+                return idGenerator;
+            }
+
+            return (ConversationIdStrategyContext _, out string id) =>
```

Comment:
Having to create a null delegate just so we can fall through and get to the "real" default behavior of using CombGuid.Generate() seems like a sign of a design problem to me.